File: easy_ecom/api/schemas/integrations.py
from __future__ import annotations

from pydantic import BaseModel, Field


class ChannelIntegrationResponse(BaseModel):
    channel_id: str
    provider: str
    display_name: str
    status: str
    external_account_id: str
    verify_token_set: bool
    inbound_secret_set: bool
    config: dict[str, str]
    created_at: str
    updated_at: str
    last_inbound_at: str | None


class ChannelIntegrationListResponse(BaseModel):
    items: list[ChannelIntegrationResponse]


class ChannelIntegrationCreateRequest(BaseModel):
    provider: str = Field(pattern="^(whatsapp|messenger|webhook)$")
    display_name: str = Field(min_length=1, max_length=255)
    external_account_id: str = Field(default="", max_length=255)
    status: str = Field(default="inactive", pattern="^(inactive|active|disabled)$")
    verify_token: str = Field(default="", max_length=255)
    inbound_secret: str = Field(default="", max_length=255)
    config: dict[str, str] = Field(default_factory=dict)


class ChannelIntegrationPatchRequest(BaseModel):
    display_name: str | None = Field(default=None, min_length=1, max_length=255)
    external_account_id: str | None = Field(default=None, max_length=255)
    status: str | None = Field(default=None, pattern="^(inactive|active|disabled)$")
    verify_token: str | None = Field(default=None, max_length=255)
    inbound_secret: str | None = Field(default=None, max_length=255)
    config: dict[str, str] | None = None


class ChannelMessageResponse(BaseModel):
    message_id: str
    conversation_id: str
    channel_id: str
    direction: str
    external_sender_id: str
    provider_event_id: str
    message_text: str
    content_summary: str
    occurred_at: str
    outbound_status: str


class ChannelMessageListResponse(BaseModel):
    items: list[ChannelMessageResponse]


class InboundEventRequest(BaseModel):
    provider_event_id: str = Field(default="", max_length=255)
    sender_external_id: str = Field(min_length=1, max_length=255)
    sender_name: str | None = Field(default=None, max_length=255)
    message_text: str | None = Field(default=None, max_length=2000)
    occurred_at: str | None = Field(default=None, max_length=64)
    metadata: dict[str, str] = Field(default_factory=dict)


class InboundIngestResponse(BaseModel):
    accepted: bool
    client_id: str
    channel_id: str
    conversation_id: str
    message_id: str


class ConversationResponse(BaseModel):
    conversation_id: str
    channel_id: str
    external_sender_id: str
    status: str
    customer_id: str | None
    linked_sale_id: str | None
    last_message_at: str
    updated_at: str


class ConversationListResponse(BaseModel):
    items: list[ConversationResponse]


class ConversationDetailMessage(BaseModel):
    message_id: str
    direction: str
    message_text: str
    content_summary: str
    occurred_at: str
    outbound_status: str


class ConversationDetailResponse(BaseModel):
    conversation_id: str
    channel_id: str
    external_sender_id: str
    status: str
    messages: list[ConversationDetailMessage]


class OutboundPrepareRequest(BaseModel):
    channel_id: str = Field(min_length=1, max_length=64)
    conversation_id: str = Field(min_length=1, max_length=64)
    message_text: str = Field(min_length=1, max_length=2000)
    recipient_external_id: str = Field(min_length=1, max_length=255)
    metadata: dict[str, str] = Field(default_factory=dict)


class OutboundPrepareResponse(BaseModel):
    dispatch_intent_id: str
    status: str
    provider: str
    delivery_deferred: bool
    deferred_reason: str
    ai_context_hint: dict[str, object] | None

File: frontend/types/integrations.ts
export type ChannelIntegration = {
  channel_id: string;
  provider: 'whatsapp' | 'messenger' | 'webhook';
  display_name: string;
  status: 'inactive' | 'active' | 'disabled';
  config_saved: boolean;
  webhook_key?: string;
  external_account_id: string;
  phone_number_id: string;
  phone_number: string;
  verify_token_set: boolean;
  inbound_secret_set: boolean;
  access_token_set: boolean;
  webhook_verified_at: string | null;
  last_webhook_post_at: string | null;
  signature_validation_ok: boolean | null;
  graph_auth_ok: boolean | null;
  outbound_send_ok: boolean | null;
  openai_ready?: boolean;
  openai_probe_ok: boolean | null;
  last_error_code: string | null;
  last_error_message: string | null;
  last_provider_status_code: number | null;
  last_provider_response_excerpt: string | null;
  last_diagnostic_at: string | null;
  next_action: string;
  default_location_id: string | null;
  auto_send_enabled: boolean;
  agent_enabled: boolean;
  model_name: string;
  persona_prompt: string;
  config: Record<string, string>;
  created_at: string;
  updated_at: string;
  last_inbound_at: string | null;
  last_outbound_at: string | null;
};

export type ChannelMessage = {
  message_id: string;
  conversation_id: string;
  channel_id: string;
  direction: 'inbound' | 'outbound';
  external_sender_id: string;
  provider_event_id: string;
  message_text: string;
  content_summary: string;
  occurred_at: string;
  outbound_status: string;
};

export type ChannelConversation = {
  conversation_id: string;
  channel_id: string;
  external_sender_id: string;
  status: string;
  customer_id: string | null;
  linked_sale_id: string | null;
  last_message_at: string;
  updated_at: string;
};

export type WhatsAppMetaIntegrationPayload = {
  display_name: string;
  external_account_id: string;
  phone_number_id: string;
  phone_number: string;
  verify_token: string;
  access_token: string;
  app_secret: string;
  default_location_id?: string | null;
  auto_send_enabled: boolean;
  agent_enabled: boolean;
  model_name: string;
  persona_prompt: string;
};

export type WhatsAppMetaIntegrationResult = {
  channel: ChannelIntegration;
  setup_verify_token: string | null;
};

export type ChannelDiagnostics = {
  config_saved: boolean;
  verify_token_set: boolean;
  webhook_verified_at: string | null;
  last_webhook_post_at: string | null;
  signature_validation_ok: boolean | null;
  graph_auth_ok: boolean | null;
  outbound_send_ok: boolean | null;
  openai_ready: boolean;
  openai_probe_ok: boolean | null;
  last_error_code: string | null;
  last_error_message: string | null;
  last_provider_status_code: number | null;
  last_provider_response_excerpt: string | null;
  last_diagnostic_at: string | null;
  next_action: string;
};

export type ChannelDiagnosticsEnvelope = {
  diagnostics: ChannelDiagnostics;
  provider_details: Record<string, string | number | null>;
};

export type ChannelSmokePayload = {
  recipient: string;
  text: string;
};

export type ChannelSmokeResult = {
  ok: boolean;
  provider_event_id: string | null;
  message: string;
  diagnostics: ChannelDiagnostics;
  provider_details: Record<string, string | number | null>;
};

export type ChannelRunDiagnosticsResult = {
  channel: ChannelIntegration;
  diagnostics: ChannelDiagnostics;
  provider_details: Record<string, string | number | null>;
};

export type ChannelLocation = {
  location_id: string;
  name: string;
  is_default: boolean;
};
